src/main/java/net/minecraftforge/event/AddReloadListenerEvent.java

```diff
@@ -22,33 +22,72 @@
 
 /**
  * The main ResourceManager is recreated on each reload, just after {@link ReloadableServerResources}'s creation.
+ * A copy of the listeners attached from vanilla can be obtained from {@link ReloadableServerResources#listeners()}
  *
  * The event is fired on each reload and lets modders add their own ReloadListeners, for server-side resources.
  * The event is fired on the {@link MinecraftForge#EVENT_BUS}
  */
 public class AddReloadListenerEvent extends Event
 {
+    @Deprecated(forRemoval = true, since = "1.19.2")
     private final List<PreparableReloadListener> listeners = new ArrayList<>();
+    private final List<PreparableReloadListener> returnList;
     private final ReloadableServerResources serverResources;
 
-    public AddReloadListenerEvent(ReloadableServerResources serverResources)
+    public AddReloadListenerEvent(ReloadableServerResources serverResources, List<PreparableReloadListener> returnList)
     {
         this.serverResources = serverResources;
+        this.returnList = returnList;
     }
```

Comment:
_Technically_, this is a breaking change since this is not yet marked as internal API.

To remedy this, we need a constructor with the original parameter marked as deprecated for removal, and a constructor with the new parameters list marked with `@ApiStatus.Internal`.

Reply:
I've seen PRs that change event constructors like this been accepted before, sometimes I hear comments like this, sometimes I don't, what decides what the rules are?

Reply:
Formal rules on this are yet to be written down, but we have come to some consensus (at least within the team) on this in the past weeks/months. 

The main idea is that we cannot break public API at this point (because we're past an RB), and unless otherwise declared, anything `public` is part of the public API. Some of the newer event constructors have been declared as `@ApiStatus.Internal`, bringing them out of the public API and therefore outside of the backwards compatibility promise.

Here, we are retaining the old public API event constructor and marking it for deletion to preserve backwards compatibility in the public API, while creating a new _internal_ event constructor (which we can modify later at our leisure, as long as the old constructor still kinda works).